src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Debugging/RazorLanguageService`IVsLanguageDebugInfo.cs

```diff
@@ -97,6 +115,15 @@ public int GetProximityExpressions(IVsTextBuffer pBuffer, int iLine, int iCol, i
 
     public int ValidateBreakpointLocation(IVsTextBuffer pBuffer, int iLine, int iCol, TextSpan[] pCodeSpan)
     {
+        if (!_lspServerActivationTracker.IsActive)
+        {
+            // We can't do anything if our LSP server isn't up and running, and can't initialize here due to UI thread dependency issues
+            // Returning like this means the debugger will place a breakpoint in the margin, but not highlight a span in the line.
+            // We trust that it will later validate the breakpoint location and remove it if it's not valid, and that validation
```

Comment:
> and that validation happens via LSP anyway

❓why do we do this at all then?

Reply:
It's a much nicer user experience to have the breakpoints placed correctly from the start, and matched the C# experience.

Reply:
Is the LSP validation that much slower? That seems like a problem. I might be missing some context here though, I haven't looked at LSP breakpoint methods

Reply:
Oh no, sorry. The debugger currently only supports COM for initial breakpoint placement, and then will validate existing breakpoints via LSP when edits are made, and when debugging starts etc.

Reply:
Ah, my mental model was wrong :) Thanks for clearing it up